pages/dkp/kaptain/2.2.0/sdk/1.3.x/api/index.md

```diff
@@ -0,0 +1,1200 @@
+- `timeout`: Number of seconds to wait for the experiment to complete before timing-out.
+
+**Returns**:
+
+True if successful, otherwise False
+
+<a name="kaptain.model.models.Model.deploy"></a>
+#### deploy
+
+```python
+ | deploy(model_uri: Optional[str] = None, autoscale: int = 2, gpus: Optional[int] = None, cpu: Optional[str] = None, memory: Optional[str] = None, resources: Optional[Resources] = None, replace: bool = False, transformer: Optional[Transformer] = None, **kwargs: str, ,) -> bool
+```
+
+Deploys a model.
+
+##### Simple / advanced resource API
+
+Resources may be specified via the 'simple' resource parameters::
+
+model.deploy(model_uri=uri, cpu=1, memory="2G", gpus=0)
+
+... the deployed model process will have both the request and limit set for all resource parameters.
+
+More fine-grained resource specification is possible via the 'resources' parameter::
+
+model.deploy(model_uri=uri, resources=Resources(cpu_request=1, memory_limit="2G", gpu_limit=gpus))
+
+It is illegal to specify both the 'resources' parameter or any 'simple' resource parameters (gpus, memory, cpu).
+
+**Arguments**:
+
+
+- `model_uri`: URI of the saved model to be loaded. If None, the default location managed by Kaptain is chosen based on the most recent state of the model.
+- `autoscale`: Target concurrency (default: 2).
+- `gpus`: Number of GPUs to use (default: 0).
+- `memory`: Amount of memory for each worker (optional),
+- `cpu`: Number of CPUs to use for each worker (optional).
+- `resources`: Advanced API for resource specification. Do not use in tandem with the parameters gpus, memory and cpu (optional).
+- `replace`: Safety flag to avoid accidental redeployment of the model. If True, the previously deployed model will be replaced. If False, an error will be logged in case the model had been previously deployed.
+- `transformer`: Transformer is an component which does pre/post processing alongside with model inference. It usually takes raw input and transforms them to the input tensors model server expects.
+- `kwargs`: Keyword arguments for the deployment.
+
+**Returns**:
+
+True if successful, otherwise False
+
+<a name="kaptain.model.models.Model.deploy_canary"></a>
+#### deploy\_canary
+
+```python
+ | deploy_canary(canary_traffic_percentage: int, model_uri: Optional[str] = None, **kwargs: str, ,) -> None
+```
+
+Deploys a model in a canary with a pre-determined percentage of traffic.
+A canary deployment allows a model to be run in parallel with a baseline or previous model
+revision. This allows traffic to be split, so the latest revision can be checked for
+possible issues with model (e.g. compared to the baseline) or system (e.g. latency)
+performance.
+To deploy a model to the canary, a previously deployed model revision must exist.
+
+------------------------------
+To deploy canary with 30 percent traffic:
+
+model.deploy_canary(canary_traffic_percentage=30)
```

Comment:
@alembiewski  wondering if these should be code snippets as well? Or are these plain text as shown in the preview?

Reply:
these pages are auto-generated by `pydoc-markdown` from python code. Given that this is an API reference, I would say it's fine to keep it as is